### ISSUE TITLE
Running sitespeed test in pods, after few hours the pods go down

### ISSUE THREAD

**Issue by mdvaliyullah** [sitespeedio/sitespeed.io#4093]
### Your question

Hi Team,

while running sitespeed tests in pods, I observed that memory usages increases continuously and the tests were scheduled to run every 15 minutes, after few hours the pods go down, we set up the --max-old-space-size to 8GB but still the pod goes down once it reached the threshold. please advise

Thanks,
Valiyullah

**Comment by madan** [comment]
I assume you are running continually in the same pod; consider using a cronjob to run your tests so the control plane spins up a new pod for each schedule.

**Comment by mdvaliyullah** [comment]
Hi @soulgalore my use case is to run different urls list during each schedule, so i wanted to keep the container running for long period, is there way to clean up the unused memory after each run?

**Comment by soulgalore** [comment]
Ok, so do you have your own container or have you overridden the start script or how do you do it, can you share a little more?

**Comment by mdvaliyullah** [comment]
I have own container with sitespeed image. No changes to startup script, just passing the arguments details like url list, iteration when starting the test to the script.

**Comment by soulgalore** [comment]
Ok. How does it work, how do you keep the container open, isn't closed after a run?

**Comment by mdvaliyullah** [comment]
Yes it is not closed after the test run, the sitespeed image is used as a base image in the container

**Comment by soulgalore** [comment]
But you exit node? So like for each test you run, you run the start script? Do you have anything setup to monitor the memory usage, is it constant increasing?

**Comment by mdvaliyullah** [comment]
Yes, I checked there is no process running once the test completes. I run the start script for every test once the previously running test completes.
memory usage gradually increases for each test once it reaches the usage threshold the pods gets restarted.

**Comment by soulgalore** [comment]
Ok, can you get more granularity to see what uses the memory? Since you use your own container, do you have something else running in it?

**Comment by soulgalore** [comment]
Closing since we didn't get any feedback.
